### PR TITLE
resolve deque deadlock using state

### PR DESCRIPTION
**Pull request by eldpswp99**
resolve deque deadlock while iteration

by using CPython method, added state property to PyDeque.

by using state, RustPython can detect mutation while iterating.

### CODE REVIEW

File: vm/src/stdlib/collections.rs [hunk 1]
```diff
@@ -447,7 +493,8 @@ mod _collections {
     struct PyDequeIterator {
         position: AtomicCell<usize>,
         status: AtomicCell<IterStatus>,
-        length: usize, // To track length immutability.
+        length: usize,
+        // To track length immutability.
```
Comment (youknowone) [inline vm/src/stdlib/collections.rs:497]: this comment looks to be placed wrong

File: vm/src/stdlib/collections.rs [hunk 2]
```diff
@@ -171,23 +177,37 @@ mod _collections {
             PyDeque {
                 deque: PyRwLock::new(self.borrow_deque().clone()),
                 maxlen: self.maxlen,
+                state: AtomicCell::new(self.state.load()),
             }
         }
 
+        fn get_cloned_vec_deque(
```
Comment (DimitrisJim) [inline vm/src/stdlib/collections.rs:184]: I don't think this function is really necessary, calling `clone` on the lock should get you your `VecDeque` just fine.

**Review by DimitrisJim — CHANGES_REQUESTED** [review]

File: vm/src/stdlib/collections.rs (hunk 2 above)
Comment (eldpswp99) [reply vm/src/stdlib/collections.rs]: I fixed with using `clone`. Thanks!

File: vm/src/stdlib/collections.rs (hunk 1 above)
Comment (eldpswp99) [reply vm/src/stdlib/collections.rs]: I fixed this, and also place new comment about state using this placement. Thanks!

**Review by DimitrisJim — APPROVED** [review]

**Pull request merged**